TypeScript/src/pages/main-page/MainPage.tsx

```diff
@@ -5,20 +5,24 @@ import InvalidValidIndentifierPage from "../error-message/InvalidValidIdentifier
 import SettingsPage from "../settings/SettingsPage";
 import UserProfile from "../user-profile/UserProfile";
 import DefaultPage from "../default-page/DefaultPage";
+import { PageIdentityEnum } from "../../constants/PageIdentityEnum";
 
 function MainPage() {
-  const pageRoute: number = +localStorage.getItem("pageId");
+  const pageRoute: PageIdentityEnum = localStorage.getItem(
+    "pageId"
+    // going to keep the casting here for now. Moving forward we should find a solution to move away from local storage to get rid of this casting.
```

Comment:
maybe we can look into more low level operations like allocating a heap memory or something that can be used accessed later.